src/Plugin/ServiceDefinition/EntityPost.php

```diff
@@ -32,14 +34,18 @@ class EntityPost extends ServiceDefinitionEntityRequestContentBase {
   public function processRequest(Request $request, RouteMatchInterface $route_match, SerializerInterface $serializer) {
     $entity = parent::processRequest($request, $route_match, $serializer);
     if ($entity) {
-      $entity->save();
-      return [$entity->getEntityType()->id() => $entity->id()];
+      try {
+        $entity->save();
+        if ($entity->id()) {
+          drupal_set_message($this->t("Entity of type @type was created.", ['@type' => $entity->getEntityType()->id()]));
+          return $entity->toArray();
+        }
+      }
+      catch(EntityStorageException $e) {
+        throw new HttpException('500', $e->getMessage());
```

Comment:
HttpExceptions are thrown when we want manual control over what's returned at the Service response layer. In this case we're catching an EntityStorageException and passing its message through a 500 (since 500 is right, but that exception has a better reason than we're going to get from anything else).